whirlpool/appliance.py

```diff
@@ -56,7 +56,11 @@ def _event_socket_handler(self, msg):
         for attr, val in json_msg["attributeMap"].items():
             if not self.has_attribute(attr):
                 continue
-            self._set_attribute(attr, str(val), timestamp)
+            if self._check_previous_attribute_value(attr, json_msg):
+                self._set_attribute(attr, str(val), timestamp)
+            else:
+                LOGGER.debug("Previous data mismatch - fetching new dataset")
+                self.fetch_data()
```

Comment:
we cannot do this since it will do one request for each attribute.
I also don't really think it is a good approach to be making requests just because we got unchanged data. If the data is not changing, then we will keep requesting data non stop.

Reply:
It will only do one request for each attribute that is mismatched - after we do the `fetch_data()` request 1 time, the previous data will be matched for every attribute, assuming each attribute had a mismatch.